docs/hugo/content/contributing/extension-points/error-classifier.md

```diff
@@ -0,0 +1,384 @@
+    details, err := next(cloudError)
+    if err != nil {
+        return core.CloudErrorDetails{}, err
+    }
+
+    // Check for resource-specific transient errors
+    if ex.isTransientError(cloudError) {
+        details.Classification = core.ErrorRetryable
+        details.Retry = true
+        log.V(Status).Info("Marking error as retryable", "code", cloudError.Code())
+    }
+
+    return details, nil
+}
+```
+
+### Pattern 2: API Version-Specific Classification
+
+```go
+func (ex *ResourceExtension) ClassifyError(
+    cloudError *genericarmclient.CloudError,
+    apiVersion string,
+    log logr.Logger,
+    next extensions.ErrorClassifierFunc,
+) (core.CloudErrorDetails, error) {
+    details, err := next(cloudError)
+    if err != nil {
+        return core.CloudErrorDetails{}, err
+    }
+
+    // Certain errors behave differently in older API versions
+    if apiVersion == "2021-01-01" && cloudError.Code() == "ResourceQuotaExceeded" {
+        // In this API version, quota errors are temporary during provisioning
+        details.Classification = core.ErrorRetryable
+        details.Message = "Resource quota temporarily exceeded during provisioning, will retry"
+    }
+
+    return details, nil
+}
+```
+
+### Pattern 3: Enhanced Error Messages
+
+```go
+func (ex *ResourceExtension) ClassifyError(
+    cloudError *genericarmclient.CloudError,
+    apiVersion string,
+    log logr.Logger,
+    next extensions.ErrorClassifierFunc,
+) (core.CloudErrorDetails, error) {
+    details, err := next(cloudError)
+    if err != nil {
+        return core.CloudErrorDetails{}, err
+    }
+
+    // Provide more context for common user errors
+    if cloudError.Code() == "InvalidParameterValue" {
+        if strings.Contains(cloudError.Message(), "sku") {
+            details.Message = fmt.Sprintf(
+                "Invalid SKU specified: %s. "+
+                "Available SKUs for this resource: %s. "+
+                "See documentation: https://docs.microsoft.com/...",
+                extractSKU(cloudError.Message()),
+                strings.Join(ex.getValidSKUs(), ", "))
+        }
+    }
+
+    return details, nil
+}
+```
+
+### Pattern 4: Conditional Fatal Classification
+
+```go
+func (ex *ResourceExtension) ClassifyError(
+    cloudError *genericarmclient.CloudError,
+    apiVersion string,
+    log logr.Logger,
+    next extensions.ErrorClassifierFunc,
+) (core.CloudErrorDetails, error) {
+    details, err := next(cloudError)
+    if err != nil {
+        return core.CloudErrorDetails{}, err
+    }
+
+    // Some configuration errors are unrecoverable
+    if cloudError.Code() == "InvalidConfiguration" {
+        details.Classification = core.ErrorFatal
+        details.Retry = false
+        details.Message = fmt.Sprintf(
+            "Configuration error: %s. This requires manual intervention.",
+            cloudError.Message())
+    }
+
+    return details, nil
+}
+```
+
+### Pattern 5: Multiple Error Conditions
```

Comment:
This isn't really a separate pattern this is just pattern 1 again, isn't it?

Reply:
Yes. Will remove this pattern.